api/agent/drivers/docker/docker_client.go

```diff
@@ -43,6 +45,7 @@ type dockerClient interface {
 	Stats(opts docker.StatsOptions) error
 	Info(ctx context.Context) (*docker.DockerInfo, error)
 	LoadImages(ctx context.Context, filePath string) error
+	RecordWaitContainerResult(ctx context.Context, exitCode int)
```

Comment:
this API method should map directly onto an API from `fsouza/go-dockerclient` (I don't believe that it does?), it should be a method on the specific driver, not the client.